src/main/java/com/solidbrain/russads/AutomaticTaskCreator.java

```diff
@@ -0,0 +1,131 @@
+package com.solidbrain.russads;
+
+import static com.solidbrain.russads.configuration.ConfigurationHelper.AUTO_SALES_REP_KEY;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import com.getbase.Client;
+import com.getbase.models.Account;
+import com.getbase.models.Address;
+import com.getbase.models.AssociatedContact;
+import com.getbase.models.Contact;
+import com.getbase.models.Deal;
+import com.getbase.models.Lead;
+import com.getbase.models.LossReason;
+import com.getbase.models.Note;
+import com.getbase.models.Pipeline;
+import com.getbase.models.Source;
+import com.getbase.models.Stage;
+import com.getbase.models.Tag;
+import com.getbase.models.Task;
+import com.getbase.models.User;
+import com.getbase.services.DealsService;
+import com.getbase.sync.Meta;
+import com.getbase.sync.Sync;
+import com.solidbrain.russads.configuration.ConfigurationHelper;
+import com.solidbrain.russads.configuration.ConfigurationPropertyRepository;
+import com.solidbrain.russads.configuration.model.ConfigurationProperty;
+
+@Configuration
+@EnableScheduling
+public class AutomaticTaskCreator {
+
+	@Autowired
+	ConfigurationHelper configurationHelper;
+
+	@Autowired
+	ConfigurationPropertyRepository dao;
+
+	@Value("${BASECRM_ACCESS_TOKEN}")
+	private String BASECRM_ACCESS_TOKEN;
+
+	@Value("${BASECRM_DEVICE_UUID}")
+	private String BASECRM_DEVICE_UUID;
+
+	private final DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
+	private Client client;
+
+	private List<ConfigurationProperty> salesRepIdsWithAutomaticTasks;
+
+	@PostConstruct

```

Comment:
When we have constructor-based injection we do not need `@PostConstruct`. And `@PostConstruct` again breaks POJO style of classes. In this example if you just create an instance of the class it will not work because noone is going to call this method.